### ISSUE TITLE
Option to redirect from the bsky.social handle to a custom domain instead of freeing the bsky.social handle

### ISSUE THREAD

**Issue by rubenlg** [bluesky-social/social-app#5904]
### Describe the Feature

According to the documentation: "Note: if you change your default Bluesky username (with the `.bsky.social` suffix) to a custom domain, your old username will be available for someone else to use."

This isn't ideal because of bad actors seeking to impersonate businesses. While some people might prefer freeing the old `{USERNAME}.bsky.social` ID, others who use `{OFFICIAL_COMPANY_NAME}.bsky.social` might prefer holding onto their original ID to avoid impersonation.

Could we add an option to preserve and redirect the original `.bsky.social` username when switching to a custom domain? This would let users choose whether to release or retain their original username, helping prevent impersonation while still giving flexibility to those who want to free up their old username.

### Describe Alternatives

The alternative is to create a separate account after transitioning to the custom domain, for the sole purpose of reserving the ID, maybe with a profile description that points to the custom domain. But it sounds wasteful in terms of resources.

**Comment by madeindex** [comment]
This seems like a big issue that many people are interested in!

There are so many questions the feature leaves one asking.

Such as, what would happen to the link to this issue on Bluesky? https://bsky.app/profile/madeindex.bsky.social/post/3lacl6scs4i26

After a name change, the old name or links should simply 301 redirect to the new ones. That is also the method recommended by the [World Wide Web Consortium (W3C)](https://www.w3.org/QA/Tips/reback)

**Comment by trivikr** [comment]
I know folks who created BlueSky accounts before custom domains were available, who are reluctant to migrate to avoid their existing `<handle>.bsky.social` to be impersonated.

Some of them are choosing to create a dummy account which follows the custom domain to avoid impersonation. But that shows two BlueSky accounts with the same name in search results.

I feel the redirect from the bsky.social handle to a custom domain should happen automatically when someone migrates.

**Comment by chylex** [comment]
If this is implemented, what happens with the existing dummy accounts? I just created an account, and I don't know if it's better to wait for the redirect, or create a dummy account now with the expectation that I'll be able to merge the accounts later.

**Comment by Tamschi** [comment]
This is somewhat relevant: #5303
It seems like the team is at least aware of the issue.

**Comment by volcanodiscovery** [comment]
It seems an important (missing) feature to me. I would propose that every account gets a <handle>.bsky.social handle, which always serves as the primary identifier and always resolves to this account, while a custom-domain handle should be an optional alias that is simply shown instead and can be used to resolve to the original handle.

**Comment by Tamschi** [comment]
That's not necessary, there already is a stable identifier (usually) of the form `did:plc:…`.
That's surfaced sometimes when you use the share link feature on Bluesky, since that's how you make permalinks on Bluesky.

What's needed is 'just' reserving the second handle and having it continue forward to the account.
(Of course this is a quite bit more complicated in practice than me writing that down, but doesn't sound terribly difficult at least for someone already familiar with the system.)

**Comment by climatebrad** [comment]
This is a very important safety and security feature.

**Comment by trivikr** [comment]
I'm also holding on my vanity.bluesky.social handle.

Post a skeet on your BlueSky account requesting followers who would like this feature to 👍 .
That will give data to BlueSky team to prioritize this feature request.

**Comment by janaka** [comment]
also just found this https://bsky.app/profile/jacob.gold/post/3kh6re46yd42k

**Comment by volcanodiscovery** [comment]
> also just found this https://bsky.app/profile/jacob.gold/post/3kh6re46yd42k
Unfortunately, this isn't a solution: Setting CNAME on the domain seems to require to actually use the same custom (sub)domain as handle already and giving up on the bsky handle.

**Comment by trusktr** [comment]
This would be a nice feature.

For now the workaround is to create a second account with the bsky.app username but it requires a second email address. For example:

- https://bsky.app/profile/joe.trusktr.io
- https://bsky.app/profile/trusktr.bsky.social

Do it as soon as you switch to a new domain username, and most likely someone won't snipe your username, unless you're so popular that someone set up a bot to snipe it as soon as you release it.

Having an official option will prevent people/bots from sniping the name.

**Comment by trusktr** [comment]
Another good reason to have this feature would be that, if people mentioned you with your old `bsky.social` username before you migrated to a custom domain username, we'd want those links to still go to the same user.

For example it would be great if after someone migrated from `@foo.bsky.social` and `@foo.domain`, both would link to the same profile, and any posts that mentioned my old username would not be broken.

**Comment by Tamschi** [comment]
> Another good reason to have this feature would be that, if people mentioned you with your old `bsky.social` username before you migrated to a custom domain username, we'd want those links to still go to the same user.
> 
> For example it would be great if after someone migrated from `@foo.bsky.social` and `@foo.domain`, both would link to the same profile, and any posts that mentioned the old username would not be broken.

That already works. The mentions use DIDs internally, which don't change when the handle does.

**Comment by volcanodiscovery** [comment]
But it would not work for links from outside that use the old handle. I think there's no way around other than keeping the bsky handle connected as failover once the user changes it to a custom domain.

**Comment by codybrom** [comment]
I appreciate the thoughtful perspectives shared here and the valid concerns being raised. That said, I’d like to respectfully offer a different take.

Former handle reservation, especially in the federated, multi-server future envisioned for the AT Protocol, feels like quicksand, with potential for unintended long-term consequences. While the intent to prevent impersonation is understandable, allowing custom domain users to also hold an additional .bsky.social handle risks repeating the same mistakes seen on early Twitter and other centralized platforms. Without strong guardrails, Twitter’s single namespace enabled early users and parody accounts to take advantage of notable names, leaving many unavailable to their expected owners. But handles alone are inherently imperfect, and anyone can create similar or derivative alternatives, which on Twitter lead to the proliferation of @real____ and @official____ handles that, ironically, some may want to preserve when transitioning to Bluesky.

The AT Protocol’s support for custom domain handles is a clear and robust solution, and I don't think anyone in this issue disagrees. Tying identity to a domain allows users to establish unique, verifiable identities on any server without needing to reserve handles across the ecosystem, so the only real challenge is when an account moves between domains. But in the case of reserving your previous non-custom name, isn't that ultimately a temporary issue?

Mastodon has already demonstrated that tools like rel=me verification and domain-linked handles work well without requiring users to constantly chase handle reservations across new instances.

If we allow for passive handle reservations now, what about when additional AT Protocol servers pop up? By supporting some amount of official handle squatting, aren't we encouraging mass name-squatting when new AT Protocol servers emerge? What about instances that exist to squat handles en masse? The solution isn't replicating the past with long-term handle squatting, but better verification.

From a UX perspective, if you search for a notable name and get multiple results, the solution shouldn't be finding which one ends in `.bsky.social`, but which one offers the clearest “verification” that it is who it says it is.

For this reason, it's my opinion that more priority should be given to additional verification features and benefits that enhance trust in searches and profiles. A public record of former handles, for example, could allow labelers to clearly indicate when a handle had previously moved to a new DID or re-registered by a new account. Labelers could also flag when the same or similar handles existed across multiple domains, providing additional clarity and confidence for users. Another option could be a form of social verification, where if two accounts follow each other, they can also trade private keys offline to mark the other as a verified connection, with an ability to  display verified connections on their profiles.

I know this perspective differs from others here, but I hope it adds value to the conversation. The AT Protocol’s strength lies in its openness and scalability, and it’s critical to embrace those advantages rather than carry forward the baggage of the past.

**Comment by janaka** [comment]
@codybrom thanks for sharing these perspectives. I didn't think of the _additional AT Protocol servers popup_ scenario because I'm not very familiar with the AT Protocol. Therefore obviously cannot comment - this might give me a bit of incentive to look up this aspect :) . Definietly appreciate the learn from other platform mistakes view. My vanity driven want to hold by bsky.social handle is definitely influenced by how handle registration on those social platforms have panned out. this does include domains. I want janaka.com but it's not been available for the last 25yrs. The person who has it isn't even a Janaka though to be fair isn't squatting, there's some logic to him registering it in the past. The point, Internet domain names are the grand daddy of squatting. So from my naive understanding it doesn't solve the problem anyway. So bsky would be just transferring the problem elsewhere anyway - and i don't know how far bsky should go to solve this either. Again, I'm taking the pure vanity view of this but do appreciate the instant verifiability that comes with a domain for products and organisations. But less so for individuals.

**Comment by codybrom** [comment]
> "Internet domain names are the grand daddy of squatting. So from my naive understanding it doesn't solve the problem anyway. So bsky would be just transferring the problem elsewhere anyway..."

You're so right. Domain squatters are absolutely the _OGs_ of this, but you're bit about "transferring the problem" points to an important lesson about that fact.  

Like the AT Protocol, the [Domain Name System](https://en.wikipedia.org/wiki/Domain_Name_System) (aka DNS) is decentralized, and while domain squatting is a challenge, it hasn’t broken the Internet either. Instead, everything remains functional **_because_** it doesn’t depend on a single namespace or central authority to assign names. 

Even though ICANN, the coordinating body for DNS, oversees the root namespace and ensures the system’s stability, it doesn't micromanage individual disputes or ownership claims. Instead, it [delegates authority to registrars](https://www.icann.org/resources/pages/help/dndr/udrp-en) and [dispute resolution providers](https://www.icann.org/resources/pages/providers-6d-2012-02-25-en), who handle thousands of cases each year. These arbitration systems, which are paid for by the disputing parties, resolve conflicts over intellectual property and domain ownership without ICANN itself ever stepping in. This decentralized approach allows users and organizations to establish trust and resolve disputes without a central authority controlling every decision.

Bluesky should follow the same principle. Trying to guarantee permanent ownership of extra `.bsky.social` handles risks turning it into even more of a default namespace than it already is, centralizing the ecosystem and making it harder for additional servers and services to ever successfully launch and grow. Instead, more focus needs to be on verification and transparency, so users can establish trusted identities without a handle availability bottleneck. That way, if a handle ever  changes hands, as long as each users is easily verifiable, it won’t really matter who has what handle. This is how a system stays decentralized, scalable, and fair while avoiding the pitfalls of centralized platforms.

**Comment by janaka** [comment]
yeah, agree with your point about delegated authority/management throughout in DNS is definitely better than entirely centralised. And I do like how that works with DNS, it's neat. DNS also doesn't prevent you from having multiple records pointing at the same resource. Of course the resource needs to accept each also for it to work (mostly).

**Comment by rubenlg** [comment]
This feature doesn't make handle squatting any easier. You still need to create a Bluesky account before you can move it to a custom domain.

The real concern here is impersonation. Here's why: While Bluesky's internal system keeps all the platform links working properly when someone changes domains, that doesn't help with external links from across the internet. Think about a celebrity or known organization that's been using a `bsky.social` handle for months - their account has probably been linked from countless blogs, news sites, Wikipedia pages, and other social networks, to name a few. Some of these links can be updated easily, but many can't.

That's where the real value lies for impersonators. It's not just about having a handle that reads `famous_person.bsky.social` - it's about controlling where all those external internet links lead to.

Given that handle reservation doesn't solve anything for fresh accounts, one option might be to add a minimum time requirement before allowing someone to keep their bsky.social handle when moving to a custom domain. After all, if an account has only existed for an hour, there's no real risk of external links breaking (or becoming targets for impersonators seeking that traffic). This would help prevent people from quickly creating accounts just to reserve handles before switching domains.

Does this make sense?

**Comment by corywright** [comment]
As I see it, there are two primary issues with being unable to retain the original username/handle:

First, as we know, after your handle is released imposters can claim it. For example, as of this writing, even though @nytimes.com is now an official handle on Bluesky, nytimes.bsky.social is again available for a squatter or imposter to grab. Same for @washingtonpost.com and washingtonpost.bsky.social.  cnn.bsky.social is taken, but is it held by CNN or someone else holding on to it to later spoof CNN?

Second, for organizations who need to maintain inbound links for previously used bsky.social handles, after renaming a handle they need to then create another separate bsky.social account to claim the previous handle. This leads to unnecessary account creations, and accounts whose only purpose will be to point visitors to the new handle. This is my own reason for being involved in this issue. I created and used the superchargeinfo.bsky.social account for 16 months before deciding to change the handle to our project name, @supercharge.info. External links to the old handle need to provide a reference to the new handle, so I had to create a second account to reclaim the old handle for the sole purpose of information visitors of our new handle. This account will count towards Bluesky statistics even though it will never be used.

The ideal solution, IMO, would be to retain the original handle but have all inbound requests for the old handle redirect to the new handle.

**Comment by climatebrad** [comment]
It looks like Bluesky is in fact quietly doing something on the backend for
"special" users - if you try to create a @nytimes.bsky.social account, you
get the error "Reserved handle".

**Comment by surfdude29** [comment]
> It looks like Bluesky is in fact quietly doing something on the backend for "special" users - if you try to create a @nytimes.bsky.social account, you get the error "Reserved handle".

Yes, **nytimes** is on the list of "famous" account names that they reserved to prevent impersonation:

https://github.com/bluesky-social/atproto/blob/1e367cba2bd1ff5560c2ec5c2a5d348cd9342b65/packages/pds/src/handle/reserved.ts#L942

**Comment by janaka** [comment]
what is bsky's stance on all this? sorry, not sure who's giving their personal opinion vs bsky's stance.

having looked into atproto identity and DID I still don't understand arguments against aliasing. This is to solve the want to retain janaka.bsky.social and add janaka.dev, janaka.co.uk, one day janaka.com as alt handles to my account. I'm not talking about solving the famous_person.bsky.social squatting problem. I'm also not talking about any pre-account creation, the flip of squatting. assume bsky.social handles are first come first serve - all the other Janaka's that would want janaka.bsky.social, sorry. Just like on twitter, I was too late. 

The handle is a friendly alias for the did:plc anyway. the field is literally called `alsoKnownAs`. so far i can see some UX stuff to be thought through like which alias to display as the handle, which is the primary (one solution: let the user pick and make that the first in the list). Maybe limit to the bsky.socal handle + 1 or 2 more. But I don't get the problem with supporting handle alias.

@codybrom 
>Trying to guarantee permanent ownership of extra .bsky.social handles risks turning it into even more of a default namespace than it already is

I think i get what you are saying here. but as you say, it is the default at the moment, that ship has sailed. again to famous person / org issue is a separate issue with some common ground.

I take you point about the complication if/when a bsky2.social comes up. I personally don't expect janaka.bsky2.social to exist out of the box. then we get into the whole Mastadon and Discord type situation. Discord has had to paper over a lot of this to improve the UX.

**Comment by n0x-space-toaster** [comment]
in the [official blog post](https://bsky.social/about/blog/4-28-2023-domain-handle-tutorial) (halfway down the page) it says

> Any tags or mentions with your old handle will still point to your account.

and right below there is an update now (posting here, since last comment was Nov 26)
> Update as of December 12, 2024: If you change your default Bluesky username (with the .bsky.social suffix) to a website URL, your old .bsky.social username will be reserved for you. You no longer need to create a second account to reserve your most recent .bsky.social username. For every account, only your most recent .bsky.social username is reserved for you, and this reservation does not expire.

However there is no indication in the settings, if this is working as intended (and what the handle is/was).
Also since mentions get re-directed (quote above), why not have the `{old_handle}.bsky.social` (profile-link) do the same (if it's reserved anyways)? (which aligns pretty much with the first part of the title of this issue).

**Comment by surfdude29** [comment]
This may not quite be answering your point @n0x-space-toaster, but fwiw when you're changing your handle to a custom domain there is now text on **main.bsky.dev** telling you that your ***.bsky.social** handle will be reserved:

![IMG_2008](https://github.com/user-attachments/assets/1a56ad3d-373b-4d5c-ab1d-4bf591f7e45e)

This will be released in 1.96 soon.

> Also since mentions get re-directed (quote above), why not have the {old_handle}.bsky.social (profile-link) do the same (if it's reserved anyways)?

My understanding is that this is what should happen currently, have you seen cases where this isn't happening?

**Comment by n0x-space-toaster** [comment]
@surfdude29 i completely forgot the existence of the dev instance. But is there more to it than "just logging in there"? Because I don't see anything the like:
![grafik](https://github.com/user-attachments/assets/954fdce3-36ac-4774-a984-38e8604bf1c4)
(or would i have had to change to custom domain while on dev for this to fully work?)

 > My understanding is that this is what should happen currently, have you seen cases where this isn't happening?

https://github.com/n0x-space-toaster/.space/commit/a95403a6f9a69b62ade13f5e50a5ef52fcb1bbc2#diff-0def367113e0e846336c6a70d4182e7e4821e0c4a43249f6ca96af3088064498R34
where I changed the link from `https://n0xSpaceToaster.bsky.social` to `https://bsky.app/profile/n0x.space` (ignoring the change from subdomain-style handles to sub-page style), I would except `https://bsky.app/profile/n0xspacetoaster` to redirect to `https://bsky.app/profile/n0x.space` (this also does not work on dev for me (or again, would this work if i changed handle on dev):

![grafik](https://github.com/user-attachments/assets/a1cb8dba-f93c-41dc-8a4b-e98a71226d89)

**Comment by gaearon** [comment]
@n0x-space-toaster you're not seeing it because you already changed your handle away from `.bsky.social`, this will only work for people changing handles from now on

**Comment by Tamschi** [comment]
Would switching it back to the `.bsky.social` handle and then again to the custom one enable the reservation and redirect?

**Comment by surfdude29** [comment]
> Because I don't see anything the like:

@n0x-space-toaster Yes what @gaearon says, and if you look at the code that text is only displayed if you're currently logged in with a **.bsky.social** handle:

https://github.com/bluesky-social/social-app/blob/c339dc5b05282ba917bed9d02c914453cf38d437/src/screens/Settings/components/ChangeHandleDialog.tsx#L494-L502

> > My understanding is that this is what should happen currently, have you seen cases where this isn't happening?
> 
> [n0x-space-toaster/.space@a95403a#diff-0def367113e0e846336c6a70d4182e7e4821e0c4a43249f6ca96af3088064498R34](https://github.com/n0x-space-toaster/.space/commit/a95403a6f9a69b62ade13f5e50a5ef52fcb1bbc2#diff-0def367113e0e846336c6a70d4182e7e4821e0c4a43249f6ca96af3088064498R34) where I changed the link from `https://n0xSpaceToaster.bsky.social` to `https://bsky.app/profile/n0x.space` (ignoring the change from subdomain-style handles to sub-page style), I would except `https://bsky.app/profile/n0xspacetoaster` to redirect to `https://bsky.app/profile/n0x.space` (this also does not work on dev for me (or again, would this work if i changed handle on dev)

I thought redirects worked for situations like that, but it's not working for me either, so I guess not? 🤔

**Comment by iKevinY** [comment]
I tried spelunking through the codebase to figure out how the "reserved handle" change implemented, but wasn't successful.

Just for my own curiosity, how are "reserved handles" actually handled on the ATProto side of things? Does it just look through the changes through the `did:plc` identity changes and look for the most recent `bsky.social` alias and treat that as the reserved name? I tried looking through my [data on PDSls](https://pdsls.dev/at/did:plc:kewhcp362af66czceva4dhrl) but didn't find a blob for `ikeviny.bsky.social`, even though I confirmed my handle is reserved.

**Comment by christianaldenjacobs** [comment]
+1 that redirects aren't working after changing handle from .bsky.social to a custom domain. I had to change my handle back to my .bsky.social handle to prevent broken links to prior posts.

I will wait til Bluesky profile and post redirects work after handle updates before I move to a custom handle.

**Comment by gmtta** [comment]
There may be a good reason from a purely tech/semantic viewpoint to not keep a .bsky.social name but from a purely brand-conscious perspective, giving one up to get the other just smells funny. Considering that Bsky themselves reserved names for major media brands in advance of them joining tells me someone over there gets this.

At this point, my finger is hovering over the custom domain button but it will remain unpressed until I get to keep both the custom name, and the original .bsky.social name - and have all links - internal and external - pointing to the .bsky.social automatically transferred to the new custom domain.

The current advice to register a custom domain and then quickly (hurry!!) re-register my old name back - even though it will break all external links, is just silly. 

Sometimes the engineering/CS answer can be correct while being totally wrong.

**Comment by janaka** [comment]
> There may be a good reason from a purely tech/semantic viewpoint to not keep a .bsky.social name but from a brand-conscious perspective, giving one up to get the other just smells funny. Considering that Bsky themselves reserved names for major media brands in advance of them joining tells me someone over there gets this.
> 
> At this point, my finger is hovering over the custom domain button but it will remain unpressed until I get to keep both the custom name, and the original .bsky.social name - and have all links - internal and external - pointing to the .bsky.social automatically transferred to the new custom domain.
> 
> The current advice to register a custom domain and then quickly (hurry!!) re-register my old name back - even though it will break all external links, is just silly.
> 
> Sometimes the engineering/CS answer can be correct while being totally wrong.

agreed - I'm not changing until it's all working either. 

but some of what you are referring to isn't the latest state as per what's references here https://github.com/bluesky-social/social-app/issues/5904#issuecomment-2546741283

but it seems like the new functionality has some issues based on recent comments ^

**Comment by RobTheFiveNine** [comment]
> +1 that redirects aren't working after changing handle from .bsky.social to a custom domain. I had to change my handle back to my .bsky.social handle to prevent broken links to prior posts.
> 
> I will wait til Bluesky profile and post redirects work after handle updates before I move to a custom handle.
> 
> EDIT: Unless… is there a (expected) delay before redirects are applied? I was impatient and immediately switched back to my old handle to avoid broken links.
> 
> EDIT 2: I tried again, this time allowing more time for redirects to start working. No luck so far. https://christian.bsky.social, https://bsky.app/profile/christian.bsky.social, and posts e.g. this [example](https://bsky.app/profile/christian.bsky.social/post/3khxgvjloye2r) are all showing `Unable to resolve handle` or `Post not found` errors.

Did you first change the handle *after* they added the redirect functionality? Or did you originally change the handle before hand and then reverted to do it again?

Asking as I haven't changed my handle at all yet and willing to be a guinea pig to test it to rule out whether it is affecting people who changed their handles previously and have changed back, or if it is affecting new changes and old alike.

**Comment by gmtta** [comment]
After saying I wouldn't until I could be sure,  I changed to a custom domain AFTER the new "your original handle will be reserved..." update. I can confirm that the change was fairly immediate, and the old .bsky.social mentions within Bluesky work. 

However, external links: https://bsky.app/profile/[original_name}.bsky.social do NOT work. "unable to resolve handle" error

Additionally, as someone mentioned, I see no evidence that my original handle was saved. Clicking on the 'handle' settings after changing to a custom domain looks like the same dialog from creating a new account. I don't see anywhere where I can revert back to the original .bsky.social handle anywhere.

**Comment by madeindex** [comment]
> After saying I wouldn't until I could be sure, I changed to a custom domain AFTER the new "your original handle will be reserved..." update. I can confirm that the change was fairly immediate, and the old .bsky.social mentions within Bluesky work.
> 
> However, external links: [https://bsky.app/profile/[original_name}.bsky.social](https://bsky.app/profile/%5Boriginal_name%7D.bsky.social) do NOT work. "unable to resolve handle" error
> 
> Additionally, as someone mentioned, I see no evidence that my original handle was saved. Clicking on the 'handle' settings after changing to a custom domain looks like the same dialog from creating a new account. I don't see anywhere where I can revert back to the original .bsky.social handle anywhere.

I'm also considering switching to a custom domain, but it seems risky. 
The old link format https://bsky.app/madeindex no longer works and redirects to the login page. 
Only the link format https://bsky.app/profile/madeindex.bsky.social works.
Changing it to https://bsky.app/profile/madeindex.org could result in posts and search engine links displaying 'unable to resolve handle'? :(

**Comment by UnfoldAntwerp** [comment]
> Additionally, as someone mentioned, I see no evidence that my original handle was saved. Clicking on the 'handle' settings after changing to a custom domain looks like the same dialog from creating a new account. I don't see anywhere where I can revert back to the original .bsky.social handle anywhere.

fully agree, the UI is not sufficient in giving you a piece of mind. Also completely unclear that you have a custom domain setup or how to unlink that domain to revert.

**Comment by zeldman** [comment]
https://bsky.app/profile/zeldman.bsky.social/post/3lnnk7na2ms2o

3-part Bluesky post concerning the basic usability error created by the lack of handle redirects. 

Y’all are encouraging us to use self-verified handles (@alistapart.com instead of @alistapart.bsky.com, for instance). GREAT! We love it.

But when we do it, all the links to our content on 3rd party websites—and in people’s personal bookmarks—will break. Not so great.

And the lack of usable information on the error message only adds to the sting.

If you really want folks to jump on self-verification via our domains—and you definitely DO want that—y’all need to fix this. Thanks for listening.

**Comment by zeldman** [comment]
P.S. I’m glad y’all sensibly kept our old bsky handles for us when we switch, instead of following your original plan of allowing someone else to use our old handles. (Gee, what could go wrong there?) Appreciate the thoughtfulness in that area. If you also build a redirect function that allows old URLs to point to our updated handles, you’ll have the best thing going. If you don’t, or can’t, a lot of existing users (and there are already 33+ million of us!) will have to hang on to our non-verified handles for fear of breaking all those old URLs.

**Comment by oliora** [comment]
Same here. Not moving to my own domain handle because this will break existing links. Please please add a redirect from bsky.social handle to the custom domain one.

**Comment by Tamschi** [comment]
If I'm not mistaken, the redirect would be automatic in most clients if the DID resolution for the old handles wasn't deactivated.

That comes with the caveat that Bluesky doesn't allow non-Bsky-hosted accounts to *present* with `….bsky.social` handles, but I think that can be easily ruled out by deactivating the resolution for a DID document *iff* it declares an `#atproto_pds` that isn't at `….bsky.network` **and** its `alsoKnownAs` contains an `at://….bsky.social`.
